sql/core/src/main/scala/org/apache/spark/sql/execution/aggregate/typedaggregators.scala

```diff
@@ -99,3 +97,67 @@ class TypedAverage[IN](val f: IN => Double) extends Aggregator[IN, (Double, Long
     toColumn.asInstanceOf[TypedColumn[IN, java.lang.Double]]
   }
 }
+
+class TypedMinDouble[IN](val f: IN => Double) extends Aggregator[IN, Double, Double] {
+  override def zero: Double = Double.PositiveInfinity
```

Comment:
Hmm it seems that typedAvg (already implemented), returns a NaN, while aggregate.Min returns null. 
Aligning it with typedAvg would not be possible for minLong, as NaN is only availble for Double of course. Another possibility of course would be to wrap it in Option type, but that again is not completely in line with aggregate.Min. This is because aggregate.Min is expression based, which has built in support for null as it extends aggregate.interfaces.DeclarativeAggregate, whereas typedaggregators extend Aggregator.
Aligning this properly seems like a huge refactor. What do you think the best approach is?